patterns_test.go

```diff
@@ -79,6 +79,8 @@ var TestVectors = map[string]*testVector{
 
 func TestProtocolMatching(t *testing.T) {
 	for name, tc := range TestVectors {
+		name := name
+		tc := tc
```

Comment:
We **have to** capture the loop variables here. Otherwise, we'll only ever run a single test case (that was the reason this test was flaky).